byzcoin/contract_deferred.go

```diff
@@ -419,33 +419,7 @@ func (c *contractDeferred) VerifyDeferredInstruction(rst ReadOnlyStateTrie, inst
 // also add to the hash the instanceID of the deferred contract.
 func hashDeferred(instr Instruction, instanceID []byte) []byte {
 	h := sha256.New()
-	h.Write(instr.InstanceID[:])
-	var args []Argument
-	switch instr.GetType() {
-	case SpawnType:
-		h.Write([]byte{0})
-		h.Write([]byte(instr.Spawn.ContractID))
-		args = instr.Spawn.Args
-	case InvokeType:
-		h.Write([]byte{1})
-		h.Write([]byte(instr.Invoke.ContractID))
-		args = instr.Invoke.Args
-	case DeleteType:
-		h.Write([]byte{2})
-		h.Write([]byte(instr.Delete.ContractID))
-	}
-	for _, a := range args {
-		nameBuf := []byte(a.Name)
-		nameLenBuf := make([]byte, 8)
-		binary.LittleEndian.PutUint64(nameLenBuf, uint64(len(nameBuf)))
-		h.Write(nameLenBuf)
-		h.Write(nameBuf)
-
-		valueLenBuf := make([]byte, 8)
-		binary.LittleEndian.PutUint64(valueLenBuf, uint64(len(a.Value)))
-		h.Write(valueLenBuf)
-		h.Write(a.Value)
-	}
+	instr.hashType(h)
```

Comment:
I'm lazy: is this backward-compatible, so that instances from old `spawn`s can still be correctly verified?

Reply:
Yes the only difference I saw was the instance ID included in the hash at the end. But having the copy/pasted logic here is annoying when fixing stuff like the command name.